src/js/media-controller.js

```diff
@@ -947,7 +955,26 @@ const Delegates = {
 
   [MediaUIAttributes.MEDIA_PLAYBACK_RATE](el) {
     return getPlaybackRate(el);
-  }
+  },
+
+  [MediaUIAttributes.MEDIA_STREAM_TYPE](el) {
+    const media = el.media;
+    const duration = media.duration;
+
+    if (duration === Infinity) {
+      return StreamTypes.LIVE;
+    } else if (!isNaN(duration)) {
```

Comment:
nitpick(blocking): we should use `Number.isNaN()` if we want to keep this strict:

```js
isNaN(undefined) // true
Number.isNaN(undefined) // false
isNaN(NaN) // true
Number.isNaN(NaN) // true
```

Reply:
whoops per callout, for strictness this should actually be:
`typeof duration === 'number' && Number.isNaN(duration)` (though `Number.isNaN` vs. `isNaN` is superfluous here). Alternatively, consider changing the order of default vs. on demand checks.

Reply:
an alternative "single check" is `Object.prototype.toString.call(duration) === '[object Number]'`, via https://github.com/ljharb/is/blob/master/index.js#L437

(still better to use `Number.isNaN` because `isNaN` could have false positives)

Reply:
maybe a better check here is `Number.isFinite()`?